tests/test_urlparse.py

```diff
@@ -270,7 +268,7 @@ def test_RFC1808(self):
         self.checkJoin(RFC1808_BASE, '../../g', 'http://a/g')
 
         # "abnormal" cases from RFC 1808:
-        self.checkJoin(RFC1808_BASE, '', 'http://a/b/c/d;p?q#f')
+        # this test is failed as expected: join 'http://a/b/c/d;p?q#f', ''
```

Comment:
can this be tested?

Reply:
Yes the result of joining those is http://a/b/c/d;p?q. which does not include the fragment..

Reply:
sorry, I don't understand. I mean two questions:
- which behaviour is correct?
- can we make the test check the correct behaviour, instead of commenting is out and writing that something is "as expected" in the comment?

Reply:
The correct output from urllib test is http://a/b/c/d;p?q#f. However, the result we get from GURL is http://a/b/c/d;p?q. Let me put this test back...

Reply:
I have put this test back and we can continue discussing it!